runtime/themes/gruvbox.toml

```diff
@@ -1,89 +1,148 @@
 # Author : Jakub Bartodziej <kubabartodziej@gmail.com>
 # The theme uses the gruvbox dark palette with standard contrast: github.com/morhetz/gruvbox
 
-"attribute" = "aqua1"
-"keyword" = { fg = "red1" }
-"keyword.directive" = "red0"
-"namespace" = "aqua1"
-"punctuation" = "orange1"
-"punctuation.delimiter" = "orange1"
-"operator" = "purple1"
-"special" = "purple0"
-"variable.other.member" = "blue1"
-"variable" = "fg1"
-"variable.builtin" = "orange1"
-"variable.parameter" = "fg2"
-"type" = "yellow1"
-"type.builtin" = "yellow1"
-"constructor" = { fg = "purple1", modifiers = ["bold"] }
-"function" = { fg = "green1", modifiers = ["bold"] }
-"function.macro" = "aqua1"
-"function.builtin" = "yellow1"
-"tag" = "red1"
-"comment" = { fg = "gray1", modifiers = ["italic"]  }
+"annotation" = { fg = "fg1" }
+
+"attribute" = { fg = "aqua1", modifiers = ["italic"] }
+
+"comment" = { fg = "gray", modifiers = ["italic"] }
+"comment.block" = { fg = "gray", modifiers = ["italic"] }
+"comment.block.documentation" = { fg = "gray", modifiers = ["italic"] }
+"comment.line" = { fg = "gray", modifiers = ["italic"] }
+
 "constant" = { fg = "purple1" }
-"constant.builtin" = { fg = "purple1", modifiers = ["bold"] }
-"string" = "green1"
-"constant.numeric" = "purple1"
-"constant.character.escape" = { fg = "fg2", modifiers = ["bold"] }
-"label" = "aqua1"
-"module" = "aqua1"
-
-"diff.plus" = "green1"
-"diff.delta" = "orange1"
-"diff.minus" = "red1"
-
-"warning" = "orange1"
-"error" = "red1"
-"info" = "aqua1"
-"hint" = "blue1"
+"constant.builtin" = { fg = "purple1" }
+"constant.builtin.boolean" = { fg = "purple1" }
+"constant.character" = { fg = "aqua1" }
+"constant.character.escape" = { fg = "orange1" }
+"constant.macro" = { fg = "aqua1" }
+"constant.numeric" = { fg = "purple1" }
+"constructor" = { fg = "purple1" }
+
+"definition" = { underline = { color = "aqua1" } }
+
+"diagnostic" = { underline = { color = "orange1", style = "curl" } }
+"diagnostic.deprecated" = { modifiers = ["crossed_out"] }
+"diagnostic.error" = { underline = { color = "red1", style = "curl" } }
+"diagnostic.hint" = { underline = { color = "blue1", style = "curl" } }
+"diagnostic.info" = { underline = { color = "aqua1", style = "curl" } }
+"diagnostic.warning" = { underline = { color = "yellow1", style = "curl" } }
+# "diagnostic.unnecessary" = { modifiers = ["dim"] }  # do not remove this for future resolving
+
+"error" = { fg = "red1" }
+"hint" = { fg = "blue1" }
+"info" = { fg = "aqua1" }
+"warning" = { fg = "yellow1" }
+
+"diff.delta" = { fg = "orange1" }
+"diff.minus" = { fg = "red1" }
+"diff.plus" = { fg = "green1" }
+
+"function" = { fg = "green1" }
+"function.builtin" = { fg = "yellow1" }
+"function.call" = { fg = "green1" }
+"function.macro" = { fg = "blue1" }
+"function.method" = { fg = "green1" }
+
+"keyword" = { fg = "red1" }
+"keyword.control.conditional" = { fg = "red1" }
```

Comment:
The note about duplication applies not to just the case you resolved (it was meant as an example) but lots of others, e.g. here

Reply:
Understood, thank you)